#!/usr/bin/env python
PACKAGE = "nmpc_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("maxObstacleRepulsionForce",  double_t, 0, "maxObstacleRepulsionForce", 0.5,  0.1, 5)

gen.add("tFormationRepulGain",  double_t, 0, "tFormationRepulGain", 30.0,  0, 50)

gen.add("tGoalAttractionGain",  double_t, 0, "tGoalAttractionGain", 5.0,  0, 15)

gen.add("maxVelocityMagnitude",  double_t, 0, "maxVelocityMagnitude", 1.0,  0.2, 5.0)


gen.add("neighborDistThreshold",  double_t, 0, "neighborDistThreshold", 1,  3.5, 20)

gen.add("distanceThresholdToTarget",  double_t, 0, "distanceThresholdToTarget", 3.0,  3, 15)

gen.add("copterDesiredHeightinNED",  double_t, 0, "copterDesiredHeightinNED", -5.0,  -20.0, -4.0)




gen.add("neighborDistThresholdSIM",  double_t, 0, "neighborDistThresholdSIM", 1, 0.5, 100)

gen.add("distanceThresholdToTargetSIM",  double_t, 0, "distanceThresholdToTargetSIM", 6.0,  2.5, 20)

gen.add("copterDesiredHeightinNED_SIM",  double_t, 0, "copterDesiredHeightinNED_SIM", -5.5,  -20.0, -2.0)



gen.add("INTERNAL_SUB_STEP",  double_t, 0, "INTERNAL_SUB_STEP", 0.2,  0, 20)

gen.add("deltaT",  double_t, 0, "deltaT", 0.1,  0, 5.0)

exit(gen.generate(PACKAGE, "nmpc_planner", "nmpcPlannerParams"))
